### ISSUE TITLE
policy loss should be min？

### ISSUE THREAD

**Issue by ex-yanminmin001** [lqtrung1998/mwp_ReFT#21]
# policy gradient loss
  ratio = torch.exp(logprob - cur_old_logprob)
  pg_losses = -cur_adv[:, :-1] * ratio
  pg_losses2 = -cur_adv[:, :-1] * torch.clamp(ratio, 1.0 - 0.2, 1.0 + 0.2)
  pg_loss = ((torch.max(pg_losses, pg_losses2) * cur_mask[:, :-1]).sum(dim=-1) / resp_len_per_sample).mean()
是不是应该是：
pg_loss = ((torch.min(pg_losses, pg_losses2) * cur_mask[:, :-1]).sum(dim=-1) / resp_len_per_sample).mean()

**Comment by lqtrung1998** [comment]
Hi @ex-yanminmin001 ,

Please note that we have a "minus" sign in front of the advantage, so the objective should still use "max" operation.

Best Regards,
Trung

**Comment by huangqingyi-code** [comment]
Hi, @lqtrung1998 
# policy gradient loss
ratio = torch.exp(logprob - cur_old_logprob)
pg_losses = -cur_adv[:, :-1] * ratio
pg_losses2 = -cur_adv[:, :-1] * torch.clamp(ratio, 1.0 - 0.2, 1.0 + 0.2)
pg_loss = ((torch.max(pg_losses, pg_losses2) * cur_mask[:, :-1]).sum(dim=-1) / resp_len_per_sample).mean()
it is the same as the one below:
pg_loss = -((torch.min(pg_losses, pg_losses2) * cur_mask[:, :-1]).sum(dim=-1) / resp_len_per_sample).mean()
It seems unrelated to the "minus" sign.

**Comment by shoyua** [comment]
> Hi, @lqtrung1998
> 
> # policy gradient loss
> ratio = torch.exp(logprob - cur_old_logprob) pg_losses = -cur_adv[:, :-1] * ratio pg_losses2 = -cur_adv[:, :-1] * torch.clamp(ratio, 1.0 - 0.2, 1.0 + 0.2) pg_loss = ((torch.max(pg_losses, pg_losses2) * cur_mask[:, :-1]).sum(dim=-1) / resp_len_per_sample).mean() it is the same as the one below: pg_loss = -((torch.min(pg_losses, pg_losses2) * cur_mask[:, :-1]).sum(dim=-1) / resp_len_per_sample).mean() It seems unrelated to the "minus" sign.

"max(pg_losses, pg_losses2)" is the same as "-min(-pg_losses, -pg_losses2)"
